extension/observer/cfgardenobserver/config_test.go

```diff
@@ -0,0 +1,51 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package cfgardenobserver
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/confmap/confmaptest"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer/cfgardenobserver/internal/metadata"
+)
+
+func TestLoadConfig(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		id       component.ID
+		expected component.Config
+	}{
+		{
+			id:       component.NewID(metadata.Type),
+			expected: NewFactory().CreateDefaultConfig(),
+		},
+		{
+			id: component.NewIDWithName(metadata.Type, "all_settings"),
+			expected: &Config{
+				Endpoint:        "unix:///var/vcap/data/garden/garden.sock",
+				RefreshInterval: 20,
```

Comment:
I suspect this is an improper type setting, where the number here needs to be a duration, rather than a plain int.